mysql/rdsmysql/rdsmysql.go

```diff
@@ -87,45 +70,36 @@ func init() {
 
 // OpenMySQLURL opens a new RDS database connection wrapped with OpenCensus instrumentation.
 func (uo *URLOpener) OpenMySQLURL(ctx context.Context, u *url.URL) (*sql.DB, error) {
-	if uo.CertSource == nil {
-		return nil, fmt.Errorf("rdsmysql: URLOpener CertSource is nil")
-	}
-	password, _ := u.User.Password()
-	params := &Params{
-		Endpoint:  u.Host,
-		User:      u.User.Username(),
-		Password:  password,
-		Database:  strings.TrimPrefix(u.Path, "/"),
-		TraceOpts: uo.TraceOpts,
+	source := uo.CertSource
+	if source == nil {
+		source = new(rds.CertFetcher)
 	}
-	db, _, err := Open(ctx, uo.CertSource, params)
-	return db, err
-}
-
-// Open opens an encrypted connection to an RDS MySQL database.
-//
-// The second return value is a Wire cleanup function that calls Close on the
-// database and ignores the error.
-func Open(ctx context.Context, provider CertPoolProvider, params *Params) (*sql.DB, func(), error) {
-	if params.Endpoint == "" {
-		return nil, nil, fmt.Errorf("open RDS: endpoint empty")
+	if u.Host == "" {
+		return nil, fmt.Errorf("open RDS: empty endpoint")
 	}
+	password, _ := u.User.Password()
 	c := &connector{
-		provider: provider,
-		params:   *params,
-		sem:      make(chan struct{}, 1),
-		ready:    make(chan struct{}),
+		addr:     u.Host,
+		user:     u.User.Username(),
+		password: password,
+		dbName:   strings.TrimPrefix(u.Path, "/"),
+		// Make a copy of TraceOpts to avoid caller modifying.
+		traceOpts: append([]ocsql.TraceOption(nil), uo.TraceOpts...),
```

Comment:
Why do we need to prevent this being modified particularly? Other parts can be modified, too.

Reply:
The slice contains a reference, so the backing array could be mutated before the connection is created.

Might be overly defensive programming, but Hyrum's Law or whatever.